siglens/siglens
# Feature: API to implement alert silencing

# Description
Added functionality to silence alerts for a certain time period. One can set "silence_minutes" in minutes via the API(/api/alerts/silenceAlert) that will prevent notifications from being sent for that alert, while still keeping the alert active behind the scenes.

### Includes:

- New "silence_minutes" column added to Alerts table to store length of silence period
- New "/alerts/silenceAlert" API that accepts alert ID and silence period and updates the Alerts table
- Updated alert notification logic to check if alert is silenced before sending notifications

Fixes #36

# Testing
- Call PUT /alerts/silenceAlert API with an existing alert ID and a silence period in hours:
```
{
  "alert_id": "17471435-3abd-4c06-8f7d-aa3bed0a7a40",
  "silence_minutes": 15
}
```
- Verify entry added to Alerts table with silence_minutes by calling GET api/alerts/{alertId}
- Trigger alert and validate no notification sent out during silence period
- Wait for silence period to expire, trigger alert again and validate notification is sent

# Checklist:
Before marking your pull request as ready for review, complete the following.
